src/main/java/com/github/housepower/jdbc/misc/SQLLexer.java

```diff
@@ -102,8 +102,11 @@ public Number numberLiteral() {
             return Long.parseLong(signed + new String(Arrays.copyOfRange(data, start + (hasSigned ? 3 : 2), pos)), 2);
         } else if (isDouble || hasExponent) {
             return Double.valueOf(new String(Arrays.copyOfRange(data, start, pos)));
+        } else if (isHex) {
+            String signed = hasSigned ? data[start] + "" : "";
+            return Long.parseLong(signed + new String(Arrays.copyOfRange(data, start + (hasSigned ? 3 : 2), pos)), 16);
         } else {
-            return Long.decode(new String(Arrays.copyOfRange(data, start, pos)));
+            return Long.parseLong(new String(Arrays.copyOfRange(data, start, pos)));
```

Comment:
Long.decode 
Because we want  to accept the input string with binary or hex prefix.

Reply:
@sundy-li

Reply:
Yes, but there are if judges for `binary` and `hex` in previous code.

Reply:
@sundy-li  yeah, i see :3 thanks

Reply:
Long.decode("08") will throw exceptions.

So parsing from  `2020-01-01 08:00:00` will fail, that's a big bug.